react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+class ValidatedInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { password: '' };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({
+      password: e.target.value
+    });
+  }
+
+  validate() {
+    const password = this.state.password;
+    if (password === '') {
+      this.message = 'A password is required.';
+      this.icon = 'fas fa-times';
+    } else if (password.length < 8) {
+      this.message = 'Your password is too short.';
+      this.icon = 'fas fa-times';
+    } else {
+      this.message = '';
+      this.icon = 'fas fa-check';
```

Comment:
Instead of sticking things on `this` (and hiding them from React), `return` them so you can use their values in your `render()` method.